source/initial_conditions/adiabatic_boundary.cc

```diff
@@ -0,0 +1,316 @@
+     * If below the isotherm use an increase of .5 degrees per kilometer.
+     */
+
+    template <>
+    double
+    AdiabaticBoundary<3>::initial_temperature (const Point<3> &position) const
+    {
+          const std::pair<double, double> lat_long = lat_long_from_xyz_wgs84(position);
+          const double depth                       = position.norm() - radius_wgs84(lat_long.first);
+          const double isotherm_temp               = 1673.15;
+          const double surface_temp                = 273.15;
+          const double isotherm_depth              = get_isotherm_depth(lat_long.first, lat_long.second);
+          if (depth < isotherm_depth)
+          return isotherm_temp - (depth - isotherm_depth) * 0.0005;
+          else
+          return surface_temp + ( depth/isotherm_depth)*(isotherm_temp - surface_temp);
+     } 
+
+
+    template <int dim>
+    void
+    AdiabaticBoundary<dim>::declare_parameters(ParameterHandler &prm)
+    {
+         prm.enter_subsection("Initial conditions");
+         {
+           prm.enter_subsection("Adiabatic boundary isotherm");
+           {
+             prm.declare_entry("Adiabatic Boundary Filename",
+                               "adiabatic_boundary.txt",
+                               Patterns::FileName(),
+                               "Surface coordinates and depths to the 1673.15 K isotherm. Units: degrees and kilometers.");
+           }
+           prm.leave_subsection();
+         }
+         prm.leave_subsection();
+    }
+
+    template <int dim>
+    void
+    AdiabaticBoundary<dim>::parse_parameters(ParameterHandler &prm)
+    {
+        prm.enter_subsection("Initial conditions");
+        {
+          prm.enter_subsection("Adiabatic boundary isotherm");
+          {
+            adiabatic_boundary_file_name = prm.get("Adiabatic Boundary Filename");
+          }
+          prm.leave_subsection();
+        }
+        prm.leave_subsection();
+
+        std::ifstream input1(adiabatic_boundary_file_name.c_str());
+        AssertThrow (input1.is_open(),
+                     ExcMessage (std::string("Can't read from file <") + adiabatic_boundary_file_name + ">"));
+
+
+        /** Start loop with int count to calculate delta below for adiabatic boundary
+         */
+	int count = 0;
+        while (true)
+          { 
+            double latitude_iso, longitude_iso, depth_iso;
+            input1 >> latitude_iso >>longitude_iso >> depth_iso;
+            if (input1.eof())
+            break;
+            latitudes_iso.push_back(latitude_iso);
+            longitudes_iso.push_back(longitude_iso);
+            depths_iso.push_back(depth_iso);
+
+            /** Find first 2 numbers that are different to use in
+             * calculating half the difference between each position as delta.
+             */
+            if (count < 2 )
+              {
+                count ++;
+              }
+            if (count == 2)
```

Comment:
This code is certainly idiosyncratic. It reads in abbreviated form
```
  count=0;
  while(true)
  {
      ....;
      if (count<2)
       count++;
     if (count == 2)
     {
        ...
        ++count;
     }
  }
```
I find this incredibly difficult to understand. First, what does `count` actually count? Secondly, I think there ought to be a clearer way to guide control flow here.